hack/lib/catalogsource.bash

```diff
@@ -79,6 +79,14 @@ spec:
   sourceType: grpc
 EOF
 
+  # Ensure the Index pod is created with the right pull secret. The Pod's service account needs
+  # to be linked with the right pull secret before creating the Pod. This is to prevent race conditions.
+  timeout 120 "[[ \$(oc -n $OLM_NAMESPACE get pods -l olm.catalogSource=serverless-operator --no-headers | wc -l) != 1 ]]"
+  index_pod=$(oc -n "$OLM_NAMESPACE" get pods -l olm.catalogSource=serverless-operator -oname)
+  if ! oc -n "$OLM_NAMESPACE" get "$index_pod" -ojsonpath='{.spec.imagePullSecrets}' | grep dockercfg &>/dev/null; then
+    oc -n "$OLM_NAMESPACE" delete pods -l olm.catalogSource=serverless-operator
```

Comment:
Maybe I could put a little `sleep 5` before this line so as to make sure the ServiceAccount has the secret properly associate before restarting the Pod.